pkg/api/dashboard.go

```diff
@@ -369,15 +370,6 @@ func (hs *HTTPServer) RestoreDeletedDashboard(c *contextmodel.ReqContext) respon
 		return response.Error(http.StatusNotFound, "Dashboard not found", err)
 	}
 
-	guardian, err := guardian.NewByDashboard(c.Req.Context(), dash, c.SignedInUser.GetOrgID(), c.SignedInUser)
```

Comment:
This and the following guardian checks can be removed, as we're now checking the access to the resource from the middleware. Check for this particular endpoint is [here](https://github.com/grafana/grafana/pull/102314/files#diff-67a6e36014be865053c43d38b2091bd2d3f17a91b117d97c0f5244cd7d71127dR480).